packages/compass-connections/src/stores/connections-store.ts

```diff
@@ -291,6 +283,10 @@ export function useConnections({
   removeConnection: (connectionInfo: ConnectionInfo) => void;
   reloadConnections: () => void;
 } {
+  const connectionProvider = useContext(
+    ConnectionProviderContext
+  ) as ConnectionProvider;
```

Comment:
We typically provide some encapsulation to hide the exact details of the React.Context provider, including a check against null-ish/uninitialized values in the package that exports them; e.g. https://github.com/mongodb-js/compass/blob/3965d4003c0bdd6c4b32f7d053060e11c1ed54f3/packages/data-service/src/provider.ts#L18-L27

Of course that's optional, but I think it's fairly nice to get a "clean" and consistent error message essentially for free in consuming packages.